CRM/PCP/BAO/PCP.php

```diff
@@ -931,5 +931,286 @@ public static function getOwnerNotificationId($component_id, $component = 'contr
+      $values[$object->id]['pcp_page'] = "<a href='{$pageUrl}'>{$object->page}</a>";
+      $values[$object->id]['pcp_start_date'] = CRM_Utils_Date::customFormat($object->start_date);
+      $values[$object->id]['pcp_end_date'] = CRM_Utils_Date::customFormat($object->end_date);
+      $values[$object->id]['pcp_status'] = $status[$object->status_id];
+      
+      $action = array_sum(array_keys($links));
+
+      $class = '';
+
+      if ($object->status_id != $approvedId || $object->is_active != 1) {
+        $class = 'disabled';
+      }
+
+      switch ($object->status_id) {
+        case 2:
+          $action -= CRM_Core_Action::RENEW;
+          break;
+
+        case 3:
+          $action -= CRM_Core_Action::REVERT;
+          break;
+      }
+
+      switch ($object->is_active) {
+        case 1:
+          $action -= CRM_Core_Action::ENABLE;
+          break;
+
+        case 0:
+          $values[$object->id]['class'][] = 'disabled';
+          $action -= CRM_Core_Action::DISABLE;
+          break;
+      }
+
+      if (!$allowToDelete) {
+        $action -= CRM_Core_Action::DELETE;
+      }
+      
+      $values[$object->id]['action'] = CRM_Core_Action::formLink($links,
+          $action,
+          array('id' => $object->id), ts('more'), FALSE, 'pcp.selector.row', 'PCP', $object->id
+        );
+
+    }
+    return $values;
+  }
+  
+   /**
+   * @param array $params
+   *
+   * @return NULL|string
+   */
+  public static function getPcpCount(&$params) {
+    $whereClause = self::whereClause($params, FALSE);
+    $query = "SELECT 
+      COUNT(pcps.id) FROM civicrm_pcp pcps 
+      INNER JOIN civicrm_contact pcp_supporter ON pcps.contact_id = pcp_supporter.id
+      LEFT JOIN civicrm_pcp_block pcp_block ON (pcp_block.id = pcps.pcp_block_id)
+      LEFT JOIN civicrm_event event ON (event.id = pcp_block.entity_id AND pcp_block.entity_table = 'civicrm_event')   
+      LEFT JOIN civicrm_contribution_page contribution_page ON (contribution_page.id = pcp_block.entity_id AND pcp_block.entity_table = 'civicrm_contribution_page')
+      LEFT JOIN civicrm_email supporter_email ON (pcp_supporter.id = supporter_email.contact_id AND supporter_email.is_primary = 1)";
+    $query .= "
+    WHERE {$whereClause}";
+    return CRM_Core_DAO::singleValueQuery($query, $params);
+  }
+  
+  /**
+   * Generate permissioned where clause for group search.
+   * @param array $params
+   * @param bool $sortBy
+   * @param bool $excludeHidden
+   *
+   * @return string
+   */
+  public static function whereClause(&$params, $sortBy = TRUE, $excludeHidden = TRUE) {
+    $values = array();
+    $title = CRM_Utils_Array::value('title', $params);
+    if ($title) {
+      $clauses[] = "pcps.title LIKE %1";
+      if (strpos($title, '%') !== FALSE) {
+        $params[1] = array($title, 'String', FALSE);
+      }
+      else {
+        $params[1] = array($title, 'String', TRUE);
```

Comment:
Here we are mixing 'general' params like 'title' with DAO formatted params (1 => array('title', 'String'). Can't we use 2 separate arrays? Preferably returning array($whereClause, $params) rather than passing by reference. 

We haven't used this much but it's also possible to interpolate the params in-function by calling return CRM_Core_DAO::composeQuery($query, $params);